src/umap.cpp

```diff
@@ -116,35 +115,11 @@ static map<void*, _umap*> active_umaps;
 
 void* umap(void* addr, size_t length, int prot, int flags, int fd, off_t offset)
 {
-    if (!(flags & UMAP_PRIVATE) || flags & ~(UMAP_PRIVATE|UMAP_FIXED)) {
-        cerr << "umap: Invalid flags: " << hex << flags << endl;
-        return UMAP_FAILED;
-    }
-
-    flags |= (MAP_ANONYMOUS | MAP_NORESERVE);
-
-    void* region = mmap(addr, length, prot, flags, -1, offset);
-    if (region == MAP_FAILED) {
-        perror("mmap failed: ");
-        return UMAP_FAILED;
-    }
-
-    _umap *p_umap;
-    try {
-        p_umap = new _umap{region, length, fd};
-    } catch(const std::exception& e) {
-        cerr << __FUNCTION__ << " Failed to launch _umap: " << e.what() << endl;
-        return UMAP_FAILED;
-    } catch(...) {
-        cerr << "umap failed to instantiate _umap object\n";
-        return UMAP_FAILED;
-    }
-
-    active_umaps[region] = p_umap;
+    void* region = umap_mf(addr, length, prot, flags, fd, NULL, offset,0);
     return region;
 }
 //--------------------------for multi-file support----------------------
-void* umap_fits(void* addr, size_t length, int prot, int flags, int fd_num,int* fd_list,off_t offset, off_t frame)
+void* umap_mf(void* addr, size_t length, int prot, int flags, int fd_num,int* fd_list,off_t offset, off_t frame)
```

Comment:
Mostly agree, but in FITS-file case, sizes and offsets are the same, seems like a waste in case of thousands of files.